src/LightGraphsMatching.jl

```diff
@@ -1,11 +1,18 @@
-__precompile__(true)
 module LightGraphsMatching
+
 using LightGraphs
+
+using SparseArrays: spzeros
+
+using JuMP
+using MathProgBase: AbstractMathProgSolver
+import BlossomV # 'using BlossomV'  leads to naming conflicts with JuMP
+
 export MatchingResult, maximum_weight_matching, maximum_weight_maximal_matching, minimum_weight_perfect_matching
 
 """
-    type MatchingResult{T}
-        weight::T
+    type MatchingResult{U}
```

Comment:
Why `U` as type parameter?
We can also change the annotation `type -> struct`

Reply:
I took some inspiration from SimpleWeightedGraphs, where this is the type of the edge-value. I think it is confusing when we use `T` for weights, as it is usually the vertex type in `JuliaGraphs`. Not sure if `U` is the best choice though, might be better to use `W` or `WE` or `EW`.